include/aws/http/private/http2_stream_manager_impl.h

```diff
@@ -25,12 +25,19 @@ enum aws_h2_sm_connection_state_type {
 /* Live with the streams opening, and if there no outstanding pending acquisition and no opening streams on the
  * connection, this structure should die */
 struct aws_h2_sm_connection {
+    struct aws_allocator *allocator;
     struct aws_http2_stream_manager *stream_manager;
     struct aws_http_connection *connection;
     uint32_t num_streams_assigned;   /* From a stream assigned to the connection until the stream completed
                                                      or failed to be created from the connection. */
     uint32_t max_concurrent_streams; /* lower bound between user configured and the other side */
 
+    /* task to send ping periodically from connection thread. */
```

Comment:
Can we avoid adding this complexity by using TLS keepalive instead?

I know the Java SDK used HTTP PING to stop their connections from closing due to inactivity, but maybe there's another way? Can we find out which service was timing them out, try to reproduce the issue with our stream manager, and then see if TLS keepalive solves the issue?

less code is best code

Reply:
hmmm, will check with Java team. But, did a quick search.

Downside of TCP keepalive from [here](https://stackoverflow.com/questions/23238319/websockets-ping-pong-why-not-tcp-keepalive):
- It operates between proxies rather than end to end.
- As pointed out by @DavidSchwartz, it operates between TCP stacks, not between the applications so therefore it doesn’t tell us whether the application is alive or not.